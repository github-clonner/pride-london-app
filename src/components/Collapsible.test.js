// @flow
import React from "react";
import { Text } from "react-native";
import { shallow } from "enzyme";
import Collapsible from "./Collapsible";

// https://github.com/facebook/jest/issues/4359#issuecomment-413238977
jest.useFakeTimers();

it("renders correctly", () => {
  const output = shallow(
    <Collapsible>
      <Text>Hello</Text>
    </Collapsible>
  );

  expect(output).toMatchSnapshot();
});

describe("collapsing behaviour", () => {
  const render = contentHeight => {
    const output = shallow(
      <Collapsible maxHeight={200} showMoreLabel="more" showLessLabel="less">
        <Text>Hello</Text>
      </Collapsible>
    );

    const layoutEvent = output.find({ testID: "children" }).prop("onLayout");
    layoutEvent({ nativeEvent: { layout: { height: contentHeight } } });

    return output.update();
  };

  const renderAndExpand = contentHeight => {
    const collapsed = render(contentHeight);

    collapsed
      .find({ testID: "showmore" })
      .parent()
      .simulate("press");

    return collapsed.update();
  };

  describe("when content does not exceed limit", () => {
    const output = render(100);

    it("renders correctly", () => {
      expect(output).toMatchSnapshot();
    });
  });

  describe("when content exceeds limit", () => {
    const output = render(500);

    it("renders correctly", () => {
      expect(output).toMatchSnapshot();
    });

    it("shows the show more link", () => {
      expect(output.find({ testID: "showmore" }).exists()).toBe(true);
      expect(
        output
          .find({ testID: "showmore" })
          .children()
          .text()
      ).toBe("more");
    });
  });

  describe("when expanded", () => {
    const output = renderAndExpand(500);

    it("renders correctly", () => {
      expect(output.state("collapsed")).toBe(false);
      expect(output).toMatchSnapshot();
    });

    it("shows the show less link", () => {
      expect(output.find({ testID: "showmore" }).exists()).toBe(true);
      expect(
        output
          .find({ testID: "showmore" })
          .children()
          .text()
      ).toBe("less");
    });
  });
});
