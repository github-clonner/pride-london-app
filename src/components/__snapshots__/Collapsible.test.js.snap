// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`collapsing behaviour when content does not exceed limit renders correctly 1`] = `
<View>
  <View>
    <AnimatedComponent
      scrollEnabled={false}
      style={
        Object {
          "height": 100,
        }
      }
    >
      <View
        onLayout={[Function]}
        testID="children"
      >
        <Text>
          Hello
        </Text>
      </View>
    </AnimatedComponent>
  </View>
</View>
`;

exports[`collapsing behaviour when content exceeds limit renders correctly 1`] = `
<View>
  <View>
    <AnimatedComponent
      scrollEnabled={false}
      style={
        Object {
          "height": 200,
        }
      }
    >
      <View
        onLayout={[Function]}
        testID="children"
      >
        <Text>
          Hello
        </Text>
      </View>
    </AnimatedComponent>
    <AnimatedComponent
      style={
        Array [
          Object {
            "bottom": 0,
            "height": 30,
            "position": "absolute",
            "width": "100%",
          },
          Object {
            "opacity": 1,
          },
        ]
      }
    >
      <LinearGradient
        colors={
          Array [
            "rgba(255, 255, 255, 0)",
            "#ffffff",
          ]
        }
        style={
          Object {
            "bottom": 0,
            "height": 30,
            "position": "absolute",
            "width": "100%",
          }
        }
      />
    </AnimatedComponent>
  </View>
  <Touchable
    accessibilityComponentType="button"
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    delayPressIn={50}
    onPress={[Function]}
    style={
      Object {
        "alignSelf": "flex-end",
      }
    }
  >
    <TextLink
      testID="showmore"
    >
      more
    </TextLink>
  </Touchable>
</View>
`;

exports[`collapsing behaviour when expanded renders correctly 1`] = `
<View>
  <View>
    <AnimatedComponent
      scrollEnabled={false}
      style={
        Object {
          "height": 200,
        }
      }
    >
      <View
        onLayout={[Function]}
        testID="children"
      >
        <Text>
          Hello
        </Text>
      </View>
    </AnimatedComponent>
    <AnimatedComponent
      style={
        Array [
          Object {
            "bottom": 0,
            "height": 30,
            "position": "absolute",
            "width": "100%",
          },
          Object {
            "opacity": 1,
          },
        ]
      }
    >
      <LinearGradient
        colors={
          Array [
            "rgba(255, 255, 255, 0)",
            "#ffffff",
          ]
        }
        style={
          Object {
            "bottom": 0,
            "height": 30,
            "position": "absolute",
            "width": "100%",
          }
        }
      />
    </AnimatedComponent>
  </View>
  <Touchable
    accessibilityComponentType="button"
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    delayPressIn={50}
    onPress={[Function]}
    style={
      Object {
        "alignSelf": "flex-end",
      }
    }
  >
    <TextLink
      testID="showmore"
    >
      less
    </TextLink>
  </Touchable>
</View>
`;

exports[`renders correctly 1`] = `
<View>
  <View>
    <AnimatedComponent
      scrollEnabled={false}
      style={
        Object {
          "height": 0,
        }
      }
    >
      <View
        onLayout={[Function]}
        testID="children"
      >
        <Text>
          Hello
        </Text>
      </View>
    </AnimatedComponent>
  </View>
</View>
`;
